dev/Repeater/SelectionModel.cpp

```diff
@@ -619,32 +620,70 @@ void SelectionModel::SelectWithGroupImpl(int groupIndex, int itemIndex, bool sel
 void SelectionModel::SelectWithPathImpl(const winrt::IndexPath& index, bool select, bool raiseSelectionChanged)
 {
     bool selected = false;
+    bool newSelection = true;
+
+    // Handle single select differently as comparing indexpaths is faster
     if (m_singleSelect)
     {
-        ClearSelection(true /*restAnchor*/, false /* raiseSelectionChanged */);
+        if (auto const selectedIndex = SelectedIndex())
+        {
+            // If paths are equal and we want to select, skip everything and do nothing
+            if (selectedIndex.CompareTo(index) == 0 && select)
```

Comment:
switch the order of the check so we don't have to access and compare the index path if select is false.

Reply:
Great point! Fixed now